tsdb/shard.go

```diff
@@ -909,6 +930,15 @@ func (f *FieldCodec) FieldByName(name string) *Field {
 	return f.fieldsByName[name]
 }
 
+func NewFieldKeysIterator(sh *Shard, opt influxql.IteratorOptions) (influxql.Iterator, error) {
+	fn := func(m *Measurement) []string {
+		keys := m.FieldNames()
+		sort.Strings(keys)
+		return keys
+	}
+	return newMeasurementKeysIterator(sh, fn, opt)
+}
+
```

Comment:
The `newMeasurementKeysIterator` is pretty small. I'd prefer separate types but I'll leave it up to you.